warehouse/query-core/src/main/java/datawave/query/tables/async/event/VisitorFunction.java

```diff
@@ -206,29 +206,17 @@ public ScannerChunk apply(@Nullable ScannerChunk input) {
                     if (config.getSerializeQueryIterator()) {
                         serializeQuery(newIteratorSetting);
                     } else {
-                        // only expand if we have non doc specific ranges.
-                        if (!RangeDefinition.allDocSpecific(input.getRanges())) {
-                            if (!evaluatedPreviously) {
-                                // if we have an hdfs configuration, then we can pushdown large fielded lists to an ivarator
-                                if (config.getHdfsSiteConfigURLs() != null && setting.getOptions().get(QueryOptions.BATCHED_QUERY) == null) {
-                                    if (null == script)
-                                        script = JexlASTHelper.parseJexlQuery(query);
-                                    try {
-                                        script = pushdownLargeFieldedLists(config, script);
-                                        madeChange = true;
-                                    } catch (IOException ioe) {
-                                        log.error("Unable to pushdown large fielded lists....leaving in expanded form", ioe);
-                                    }
-                                }
-                            }
-                            
-                        } else {
-                            if (input.getRanges().size() == 1) {
-                                if (log.isTraceEnabled()) {
-                                    log.trace("Ensuring max pipelines is set to 1");
-                                    
+                        if (!evaluatedPreviously) {
```

Comment:
Just trying to understand this...  So in the past, if our ranges were not all doc specific (i.e. we have some shard ranges), and we haven't previously encountered this exact query plan, we will do our large fielded list work (and then later cache this remade query).  OK, cool.  I don't know why we only did this in the past when we had some shard ranges in our query.  This seems like it would be good to do all the time (per your change).  Am I following that correctly?

Reply:
That is how I read it also, but I was thinking you or @ivakegg would catch me if that was wrong

Reply:
Man, I love how this was new to me when I originally saw your PR, but now I've seen it and experienced it, and still don't know why we shouldn't ever do this for doc specific ranges.  Fun stuff.